### DISCUSSION TITLE
How to consume cairo library on Windows?

### DISCUSSION THREAD

**Discussion by Endilll** [microsoft/vcpkg#24944] [Q&A]
While on Linux one has `pkg-config` out of the box, and can leverage `FindPkgConfig`, it's quite exotic for Windows.

Adding a host-only dependency `pkgconf` resolves that, but I don't see a way to ask vcpkg for host triplet, which is required to form a path to `pkgconf` and use it to set `PKG_CONFIG_EXECUTABLE`.

I found that `vcpkg_pkgconfig_get_modules` jumps through some hoops to work around that. I tried to get advantage of that, but it seems that `x_vcpkg_pkgconfig_get_modules` is available in portfiles only.

So, what's the idiomatic way of consuming cairo and other pkg-config libraries on Windows?

**Comment by Neumann-A** [comment]
> but I don't see a way to ask vcpkg for host triplet,

Set `VCPKG_HOST_TRIPLET` and make `pkgconf` a host dependency in your manifest.

**Reply by Endilll** [reply]
I'd like to rely on host triplet auto-detection, since I'm not doing anything fancy and want to avoid human mistakes.
